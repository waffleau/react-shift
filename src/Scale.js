import React, {PropTypes, PureComponent} from 'react'

import Transition from './Transition'

export class Scale extends PureComponent {
  static defaultProps = {
    minX: 0,
    minY: 0,
    maxX: 1,
    maxY: 1
  }

  static propTypes = {
    ...Transition.baseProps,
    maxX: PropTypes.number,
    maxY: PropTypes.number,
    minX: PropTypes.number,
    minY: PropTypes.number

  }

  getFromValue() {
    const {minX, minY} = this.props
    return {
      transform: `scale3d(${minX}, ${minY}, 1)`
    }
  }

  getStyle() {
    const {style} = this.props
    return {
      transformOrigin: 'center center',
      ...style
    }
  }

  getToValue() {
    const {maxX, maxY} = this.props
    return {
      transform: `scale3d(${maxX}, ${maxY}, 1)`
    }
  }

  render() {
    const {children} = this.props
    return (
      <Transition
        {...this.props}
        style={this.getStyle()}
        easing="easeInOut"
        fromValue={this.getFromValue()}
        toValue={this.getToValue()}>
        {children}
      </Transition>
    )
  }
}

export default Scale
