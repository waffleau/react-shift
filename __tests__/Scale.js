import React from 'react'
import renderer from 'react-test-renderer'

import Scale from '../src/Scale'

test('has no scale when inactive', () => {
  const tree = renderer.create(
    <Scale active={false}>
      {'Children'}
    </Scale>
  ).toJSON()

  expect(tree).toMatchSnapshot()
})

test('has scale of 1 when active', () => {
  const tree = renderer.create(
    <Scale active>
      {'Children'}
    </Scale>
  ).toJSON()

  expect(tree).toMatchSnapshot()
})

test('can specify a minimum x scale', () => {
  const tree = renderer.create(
    <Scale
      active={false}
      minX={0.1}>
      {'Children'}
    </Scale>
  ).toJSON()

  expect(tree).toMatchSnapshot()
})

test('can specify a minimum y scale', () => {
  const tree = renderer.create(
    <Scale
      active={false}
      minY={0.1}>
      {'Children'}
    </Scale>
  ).toJSON()

  expect(tree).toMatchSnapshot()
})

test('can specify a maximum x scale', () => {
  const tree = renderer.create(
    <Scale
      active
      maxX={1.1}>
      {'Children'}
    </Scale>
  ).toJSON()

  expect(tree).toMatchSnapshot()
})

test('can specify a maximum y scale', () => {
  const tree = renderer.create(
    <Scale
      active
      maxY={1.1}>
      {'Children'}
    </Scale>
  ).toJSON()

  expect(tree).toMatchSnapshot()
})
