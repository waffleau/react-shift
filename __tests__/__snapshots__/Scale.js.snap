exports[`test can specify a maximum x scale 1`] = `
<div
  style={
    Object {
      "transform": "scale3d(1.1, 1, 1)",
      "transformOrigin": "center center",
      "transition": "transform 600ms ease-in-out",
    }
  }>
  Children
</div>
`;

exports[`test can specify a maximum y scale 1`] = `
<div
  style={
    Object {
      "transform": "scale3d(1, 1.1, 1)",
      "transformOrigin": "center center",
      "transition": "transform 600ms ease-in-out",
    }
  }>
  Children
</div>
`;

exports[`test can specify a minimum x scale 1`] = `
<div
  style={
    Object {
      "transform": "scale3d(0.1, 0, 1)",
      "transformOrigin": "center center",
      "transition": "transform 600ms ease-in-out",
    }
  }>
  Children
</div>
`;

exports[`test can specify a minimum y scale 1`] = `
<div
  style={
    Object {
      "transform": "scale3d(0, 0.1, 1)",
      "transformOrigin": "center center",
      "transition": "transform 600ms ease-in-out",
    }
  }>
  Children
</div>
`;

exports[`test has no scale when inactive 1`] = `
<div
  style={
    Object {
      "transform": "scale3d(0, 0, 1)",
      "transformOrigin": "center center",
      "transition": "transform 600ms ease-in-out",
    }
  }>
  Children
</div>
`;

exports[`test has scale of 1 when active 1`] = `
<div
  style={
    Object {
      "transform": "scale3d(1, 1, 1)",
      "transformOrigin": "center center",
      "transition": "transform 600ms ease-in-out",
    }
  }>
  Children
</div>
`;
